Userland/Libraries/LibGfx/ImageFormats/PNGLoader.cpp

```diff
@@ -1209,49 +1210,52 @@ static bool process_chunk(Streamer& streamer, PNGLoadingContext& context)
         dbgln_if(PNG_DEBUG, "Bail at chunk_size");
         return false;
     }
-    u8 chunk_type[5];
-    chunk_type[4] = '\0';
-    if (!streamer.read_bytes(chunk_type, 4)) {
-        dbgln_if(PNG_DEBUG, "Bail at chunk_type");
-        return false;
-    }
-    ReadonlyBytes chunk_data;
-    if (!streamer.wrap_bytes(chunk_data, chunk_size)) {
+
+    ReadonlyBytes chunk_type_and_data;
+    if (!streamer.wrap_bytes(chunk_type_and_data, chunk_size + 4)) {
         dbgln_if(PNG_DEBUG, "Bail at chunk_data");
         return false;
     }
+    StringView const chunk_type { chunk_type_and_data.data(), 4 };
+    ReadonlyBytes const chunk_data { chunk_type_and_data.slice(4) };
+
     u32 chunk_crc;
     if (!streamer.read(chunk_crc)) {
         dbgln_if(PNG_DEBUG, "Bail at chunk_crc");
         return false;
     }
     dbgln_if(PNG_DEBUG, "Chunk type: '{}', size: {}, crc: {:x}", chunk_type, chunk_size, chunk_crc);
 
-    if (!strcmp((char const*)chunk_type, "IHDR"))
+    if (auto const computed_crc = Crypto::Checksum::CRC32(chunk_type_and_data).digest(); computed_crc != chunk_crc) {
+        dbgln_if(PNG_DEBUG, "Checksum failed expected crc: {:x}, got: {:x}", chunk_crc, computed_crc);
+        return false;
+    }
+
+    if (chunk_type == "IHDR"sv)
```

Comment:
this down here should be a separate commit (and maybe separate PR). In fact, maybe you want to revive #17519? :)

Reply:
I do it if you give your opinion on this PR :)